### ISSUE TITLE
Comma missing in json data when exporting an object that has no material

### ISSUE THREAD

**Issue by s4e-tom** [antlafarge/ALLoader#3]
Hello,

One of our animators exported a simple object which only had meshes and animations (no material) in 3DS Max 2014 with ALLoader (master) in order to try your project.

When loading the object (using an html/js script heavily inspired from one of your exemples) we encountered an "unexpected string" error on the entire javascript code.

After a lot of research we found out that a comma was missing in the json data of our object, between the "meshes" and "animations" section. We think that exporting an object that has no material causes this problem, as adding a material and exporting again fixed the problem.

![image](https://cloud.githubusercontent.com/assets/3964857/7248602/38458038-e814-11e4-875d-77f8605b04b7.png)

**Comment by antlafarge** [comment]
Hi, thanks for the report, I will fix that soon !
Any other suggestions? :)

**Comment by s4e-tom** [comment]
Not at the moment. We've been able to display simple objects, animated and textured, without problem after figuring that out. Right now we're having another problem. We're trying to display a complex object (the json data is about 2 Mo), but nothing is displayed apart from the axis.

I have read on Stackoverflow that object loading is asynchronous in Threejs, and that rendering will fail if render() is called before loading completed. Is it certain that "(new THREE.ALLoader).load(json, mesh, textures)" finishes before calling render()? Again, I have only used Threejs with simple objects, so I have little experience and may be missing something.

![image](https://cloud.githubusercontent.com/assets/3964857/7250063/546262c6-e81f-11e4-802a-9bcce873de10.png)

**Comment by antlafarge** [comment]
Okay, don't worry about it, the callback is (normally) launched after the whole mesh load is complete. so this is not the problem.  
You should try to isolate the data. We have to know if the exported data is corrupted or not. So we should work by step. 
For example only loading geometry (without animation), and try to render the complex object as "raw" by applying to it a meshNormalMaterial.  
If this work, you can try to plug the real material but deactivating the animation in this material.  
And finally if this works, try to activate the animation.  
If you have a 3dsmax sample to send to me, don't hesitate to send by email.

**Comment by s4e-tom** [comment]
I'm afraid I'm not allowed to the you the object. I'm going to try to isolate the data with the steps you described. If I fail, I'll see if we can generate a random sample of similar complexity that reproduces the problem so you can have a look into the problem.

Thank you a lot for your help!

**Comment by antlafarge** [comment]
Yes that's what I mean.  
Okay let me know your results.

**Comment by antlafarge** [comment]
Oh okay somebody managed to do this so I added the process in the readme, but I didn't try this by myself yet.

**Comment by s4e-tom** [comment]
Hello again,

We found out that it was an absolutely normal behavior because we were only adding a single mesh to the scene (data.meshes[0]) instead of all of them. Indeed the cube you could see on the screen is a twist cube used for animations.

So our JSON contains 17 meshes: 16 twist cube, and the actual model (which also has references to the twists in "skins").
We can apply a texture to it and display it. We can also run a (slightly inacurrate) animation by adding only the model mesh to the scene and disabling the skeletonHelper update. Indeed we're having a problem with the SkeletonHelper (many javascript errors on skeletonHelper.update() ). I tried to pass the model mesh to the SkeletonHelper constructor but that didn't work. Right now I'm trying to find some documentation on that class.
 
TL;DR I overlooked a line in my code and it took me a lot of time to figure out :( also I apologize for wasting your time. Thanks for helping!

**Comment by antlafarge** [comment]
No problem, if you think there is no problem with the helper, that's great (I will fix the comma bug tonight). But if you think I can improve the exporter or loader by any way, feel free to let met know !  
And I'm interested to see the result of your final project if that's possible :)

**Comment by antlafarge** [comment]
Bug fixed
https://github.com/antlafarge/ALLoader/commit/5616304349223fb178911ed68625c3416cb4efd2
